lldb/include/lldb/Core/AddressRange.h

```diff
@@ -242,6 +244,12 @@ class AddressRange {
   lldb::addr_t m_byte_size = 0; ///< The size in bytes of this address range.
 };
 
+// Forward-declarable wrapper.
+class AddressRanges : public std::vector<lldb_private::AddressRange> {
+public:
+  using std::vector<lldb_private::AddressRange>::vector;
+};
```

Comment:
Why do you wrap this instead of using `std::vector<lldb_private::AddressRange>` below?

Reply:
I followed the same convention as here:
https://github.com/llvm/llvm-project/blob/main/lldb/include/lldb/Target/MemoryRegionInfo.h#L172

Do you have a preference?

Reply:
I see, you're following the convention for `SBMemoryRegionInfoList`. In the discussion above with Greg I proposed a different solution.

Reply:
I am not seeing what was suggested "above"? If I reverse search here for your name, I don't see anything?

Reply:
Sorry, "above" was the wrong word. I meant "below", but to be even more specific, I'm referring to this: https://github.com/llvm/llvm-project/pull/92014#discussion_r1602020217